lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1639,11 +1644,16 @@ def obj_has_parent(self, obj, parent):
             if current_parent is None:
                 return False
 
-    def select_resource_pool_by_name(self, resource_pool_name):
-        resource_pool = self.cache.find_obj(self.content, [vim.ResourcePool], resource_pool_name)
-        if resource_pool is None:
+    def select_resource_pool_by_name(self, resource_pool_name, cluster=None):
+        resource_pools = self.cache.find_obj(self.content, [vim.ResourcePool], resource_pool_name, return_list=True)
+        if resource_pools is None or len(resource_pools) == 0:
             self.module.fail_json(msg='Could not find resource_pool "%s"' % resource_pool_name)
-        return resource_pool
+        if cluster:
+            for resource_pool in resource_pools:
```

Comment:
Add condition to check like, because we don't want resource pool which is different that user specified name.

```
             if resource_pool.name != resource_pool_name:
                 continue
```